src/trace/Kconfig

```diff
@@ -35,22 +35,22 @@ config TRACEM
 config TRACE_FILTERING
 	bool "Trace filtering"
 	depends on TRACE
-	default y
+	default n
```

Comment:
This is hiding a very big change in a PR with a totally unrelated name.

Reply:
do you mean the PR subject, let me rename it. Anyway, the change is mentioned in the commit subject.

Reply:
Yes the commit is fine, I was only referring to the PR name.

This relatively complex feature took a lot of effort, so I'm pretty sure it was developed for a reason. This change could kill performance on all production systems with just one character change. So even a scary PR name may not be enough, it should probably be discussed in some larger forum / mailing list.

Reply:
> Yes the commit is fine, I was only referring to the PR name.
> 
> This relatively complex feature took a lot of effort, so I'm pretty sure it was developed for a reason. This change could kill performance on all production systems with just one character change. So even a scary PR name may not be enough, it should probably be discussed in some larger forum / mailing list.

No, in real production systems, usually the DMA trace itself is disabled thoroughly.

Reply:
The filtering is a useful developer debug feature, i.e. helps developers improve the trace SNR when debugging issues when there is high frequency trace output. However for CI usage it can hide some important details in the logs that would be useful for triage. I'm good to try this change for CI and see if it helps

Reply:
> Anyway, it filters out useful logs by mistake from time to time for me, it is legit to disable it by default to me.

Specific examples? Reproduction steps? `CONFIG_TRACE_FILTERING_ADAPTIVE` can collapse lines with different parameters but `CONFIG_TRACE_FILTERING_VERBOSITY` does not lose anything by default. They're very different but you changed both in a single sentence.

In any case these very significant change should have been submitted in a separate and more visible PR, not hidden deep down some  heap allocation PR.

The most important loss is the ability to configure per-component verbosity at run time. If the default verbosity of some components was not high enough these could have been changed without throwing the baby with the bathwater.

> for me

It is just for you then you can change your configuration in your workspace, not in everyone else's without asking.

Reply:
> > Anyway, it filters out useful logs by mistake from time to time for me, it is legit to disable it by default to me.
> 
> Specific examples? Reproduction steps? `CONFIG_TRACE_FILTERING_ADAPTIVE` can collapse lines with different parameters but `CONFIG_TRACE_FILTERING_VERBOSITY` does not lose anything by default. They're very different but you changed both in a single sentence.
> 
> In any case these very significant change should have been submitted in a separate and more visible PR, not hidden deep down some heap allocation PR.
> 
> The most important loss is the ability to configure per-component verbosity at run time. If the default verbosity of some components was not high enough these could have been changed without throwing the baby with the bathwater.
> 

@marc-hb does the change bring you or the CI bad experience? Can you list it or paste some example logs?

To me, it brings very bad user experience with removing useful/similar logs, which impact debugging e.g. from the customers captured logs:

```
[       28011.717637] (         224.062485) c0 dma-trace                 src/trace/trace.c:133  Suppressed 6 similar messages:   block %d base 0x%x size %d
[       28018.905137] (           7.187500) c0 dma-trace                 src/trace/trace.c:133  Suppressed 6 similar messages:    count %d free %d
```

The PR already stayed here > 1 month and you commented to it also @marc-hb. If you have any concern about the change itself, please say it more.

Reply:
> The PR already stayed here > 1 month and you commented to it also @marc-hb. 

Indeed I asked for a **separate PR** almost 2 months ago and you ignored that feedback and kept it in this PR.

> [       28011.717637] (         224.062485) c0 dma-trace                 src/trace/trace.c:133  Suppressed 6 similar messages:   block %d base 0x%x size %d
> [       28018.905137] (           7.187500) c0 dma-trace                 src/trace/trace.c:133  Suppressed 6 similar messages:    count %d free %d

To avoid this disabling `CONFIG_TRACE_FILTERING_ADAPTIVE` is enough, there was no need to disable `CONFIG_TRACE_FILTERING_VERBOSITY` too. Did you notice there is a difference? These are two very different features but your commit message does not make any difference between the two features.

> > No, in real production systems, usually the DMA trace itself is disabled thoroughly, 

> which impact debugging e.g. from the customers captured logs

Reply:
Yes, I just change to unselect the whole TRACE_FILTERING feature, frankly speaking I don't know how to use the _TRACE_FILTERING_VERBOSITY at all, I haven't seen the tr_dbg() logs for years. @slawblauciak asked me about this and what I shared to him is just use tr_info() only.
@marc-hb can you share how should CI and customer use this _TRACE_FILTERING_VERBOSITY today? If nobody is using it, no harm to disable it by default, no?

Reply:
@keyonjie I also disagree with your approach. Why wasn't this change a different PR? we could have fixed the 'filtering by mistake' instead of disabling the capability entirely.

@lgirdwood we keep the rule that there are TWO approvers on such critical PRs. This ignores feedback, it's not a good process.

Reply:
> @marc-hb can you share how should CI and customer use this _TRACE_FILTERING_VERBOSITY today?

By following the official documentation:
https://thesofproject.github.io/latest/developer_guides/debugability/logger/index.html#trace-filtering

Filtering by log levels and component is really the most basic feature of any logging framework.


This commit apparently caused regression #4779 , according to some people reverting it seems to make the problem go away. Of course the code should in theory work with any Kconfig combination, I am not saying changing Kconfig is the root cause. But we need the tests to pass.

Because both, unrelated settings were changed in the same commit ("just in case?"), validation could not immediately find which one setting caused the regression.

Cc: @ranj063 , @1994lwz

Reply:
These KConfig changes have been reverted in #4785 because of regression #4779. @keyonjie I now understand that `CONFIG_TRACE_FILTERING_ADAPTIVE` is the only default you wanted to change, correct? Please re-submit with only the `CONFIG_TRACE_FILTERING_ADAPTIVE` default change (it's all very brittle, so only one change at a time) and after first thoroughly trying to reproduce issue #4779 locally.

Reply:
Actually, it's very easy to turn off CONFIG_TRACE_FILTERING_ADAPTIVE at _run-time_, no recompilation needed: #5154